### DISCUSSION TITLE
The ordering of path parameters is not correct with path is a prefix and used by Depends

### DISCUSSION THREAD

**Discussion by jamesrusso** [fastapi/fastapi#11128] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated

from fastapi import APIRouter, Depends, FastAPI

app = FastAPI()
site_router = APIRouter(prefix="/site/{site_id}")

# The alert router requires a "site service" to access the alerts for that site.
alert_router = APIRouter()


def site_dep(site_id: int):
    return f"site {site_id}"


SiteDep = Annotated[str, Depends(site_dep)]


@alert_router.get("/{alert_id}")
def get_alert_by_id(
    site: SiteDep,
    alert_id: int,
):
    return {"message": f"Alert {alert_id} {site}"}


site_router.include_router(alert_router, prefix="/alert")

app.include_router(site_router)
```

### Description

* Run this fastapi application
* Visit /#docs
* You would expect that the path parameter alert_id would come second, but it comes first with site_id coming first. This order of these parameters in the openjson output reflect the order of parameters in the generated code and it's not nature to have alert_id and then site_id, it should be site_id, alert_id. 

If there is an alternate way of accomplishing this, I'm open to suggestions. Basically, all the operations are under a "site" path prefix.

### Operating System

macOS

### Operating System Details

Sonoma 14.1

### FastAPI Version

0.109.2

### Pydantic Version

2.6.1

### Python Version

Python 3.12.0

### Additional Context

<img width="1458" alt="Screenshot 2024-02-10 at 2 34 23 PM" src="https://github.com/tiangolo/fastapi/assets/682807/65b2e64f-9aac-4b84-868f-5851d8f9fc62">

**Comment by jamesrusso** [comment, marked as answer]
The workaround seems to be to include the site_id: int as the first paramter to the method in alerts, even though it's not really needed. All I need is the site service setup and configured for the correct site. 

```
@alert_router.get("/{alert_id}")
def get_alert_by_id(
    site_id: int,
    site: SiteDep,
    alert_id: int,
):```

Yields the correct ordering of the path parameters.

**Reply by YuriiMotov** [reply]
Another way is to create dependency for `alert_id`:

```python
async def alert_dep(alert_id: int):
    return alert_id


@alert_router.get("/{alert_id}")
def get_alert_by_id(
    site: SiteDep,
    alert_id: Annotated[int, Depends(alert_dep)],
):
    return {"message": f"Alert {alert_id} {site}"}

```
